feat(linter): add ignores option to overrides

xuanduc987 opened the pull request:
This will add a new `ignores` option to oxlint override config, allowing excluding some files matched by `files` from applying the override.

Resolves #15932

connorshea commented:
I think it should probably be called `ignorePatterns` to match the root-level config option name? Although I guess it depends on how we want it to work with regards to merging with the ignore values with the root ignores

xuanduc987 commented:
I chose this name to be similar to `eslint`

---

`eslint` also used to have `ignorePatterns` config option at the root level, and in the override, it had `excludedFiles` options.

But now, with flat config, `eslint` uses `ignores` name in its [configuration objects](https://eslint.org/docs/latest/use/configure/configuration-files#configuration-objects)

xuanduc987 committed:
feat(linter): add ignores option to overrides

This will add a new `ignores` option to oxlint override config, allowing
excluding some files matched by `files` from applying the override.

Resolves #15932

JasonKaz commented:
> Hey, any plans to merge this? It's in the docs but apparently it's not supported 🤔 https://oxc.rs/docs/guide/usage/formatter/config-file-reference.html#overrides-n-excludefiles

That is for oxfmt, not oxlint.

This is an essential feature used by our eslint configuration. Without it we cannot migrate to oxlint.

avcharovBetsson commented:
HI @camc314, we are in need to this feature to migrate to oxlint
 is there any issues with approving this PR?
